tests/data_context/store/test_datasource_store.py

```diff
@@ -222,3 +306,84 @@ def test_datasource_store_update_raises_error_if_datasource_doesnt_exist(
         )
 
     assert f"Unable to load datasource `{datasource_name}`" in str(e.value)
+
+
+@pytest.mark.integration
+def test_datasource_store_with_inline_store_backend_config_with_names_does_not_store_datasource_name(
+    datasource_config_with_names: DatasourceConfig,
+    datasource_config: DatasourceConfig,
+    empty_data_context: DataContext,
+) -> None:
+    inline_store_backend_config: dict = {
+        "class_name": "InlineStoreBackend",
+        "resource_type": DataContextVariableSchema.DATASOURCES,
+        "data_context": empty_data_context,
+        "suppress_store_backend_id": True,
+    }
+
+    store = DatasourceStore(
+        store_name="my_datasource_store",
+        store_backend=inline_store_backend_config,
+        serializer=YAMLReadyDictDatasourceConfigSerializer(
+            schema=datasourceConfigSchema
+        ),
+    )
+
+    key = DataContextVariableKey(
+        resource_name="my_datasource",
+    )
+
+    store.set(key=key, value=datasource_config_with_names)
+    res: DatasourceConfig = store.get(key=key)
+
+    _assert_serialized_datasource_configs_are_equal([res, datasource_config])
+
+    with open(
+        os.path.join(empty_data_context.root_directory, "great_expectations.yml")
```

Comment:
Can you use `pathlib` here?
```suggestion
        pathlib.Path(empty_data_context.root_directory) \ "great_expectations.yml"
```